src/Context/WishlistContext.jsx

```diff
@@ -0,0 +1,39 @@
+import { createContext, useContext, useReducer } from "react";
+const WishlistContext = createContext();
+const useWishlist = ()=> useContext(WishlistContext);
+
+const WishlistProvider =({children})=>{
+    function addToWishlistList(state,action){
+        const {name,_id,price,imagesrc} = action.payload;
+        let productInWishlist= false;
+        let wishlistListOnAddition= state.wishlistList.map(item=>{
+            if(item._id===action.payload._id){
+                productInWishlist=true;
+                return {...item}
+            }
+            return item;
+        })
+        if(!productInWishlist){
+            wishlistListOnAddition=[...state.wishlistList,{name,_id,price,imagesrc}];
+        }
+       return wishlistListOnAddition;
+    }
+    
+    const [state,dispatchWishlist]=useReducer((state,action)=>{
+    switch (action.type){
+        case "ADD_TO_WISHLIST":
+            return {...state, wishlistItems : state.wishlistItems + 1, price: state.price+action.payload.price,
+                totalPrice : Number(state.totalPrice) + Number(action.payload.price), wishlistList: addToWishlistList(state,action)};
+        case "REMOVE_FROM_WISHLIST":
+            return {...state, wishlistItems : state.wishlistItems - 1, price: state.price-(action.payload.price),
+                totalPrice : Number(state.totalPrice) - Number(action.payload.price), wishlistList : state.wishlistList.filter(item => item._id !== action.payload._id)}
+        default :
+            return state;
+    }
+},{wishlistItems : 0, price : 0, totalPrice: 0, wishlistList : []})
```

Comment:
Why do you need **price** && **totalPrice** in wishList ??

In the *wishlist page*, we will be displaying only the selected products, Only in the *cart page* you need to **compute** the price to **bill the products**.

And, instead of having separate variable for tracking **wishlistItems**, you can take advantage of the **wishlistList**.(i.e: wishlistList.length() => will the number of wishlistItems)

Reply:
Thank you, I have made the changes and removed wishlistItems as well because i realised it is not required either, Please review